libraries/Microsoft.Bot.Builder/Adapters/TestFlow.cs

```diff
@@ -28,14 +28,14 @@ public class TestFlow
     {
         private readonly TestAdapter _adapter;
         private readonly Task _testTask;
-        private Func<ITurnContext, Task> _callback;
+        private BotCallbackHandler _callback;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TestFlow"/> class.
         /// </summary>
         /// <param name="adapter">The test adapter to use.</param>
         /// <param name="callback">The bot turn processing logic to test.</param>
-        public TestFlow(TestAdapter adapter, Func<ITurnContext, Task> callback = null)
+        public TestFlow(TestAdapter adapter, BotCallbackHandler callback = null)
```

Comment:
I know you've already done all the work to upgrade the existing tests, but we could introduce a constructor overload here that would make things cleaner/easier for test callbacks that don't actually need to use the `CancellationToken` which I suspect would be most of them (it's actually _all_ of them currently). An overload like this could be added:

```CSharp
public TestFlow(TestAdapter adapter, Func<ITurnContext, Task> callback)
    : this(adapter, callback == null ? (BotCallbackHandler)null : (tc, ct) => callback(tc))
        {
        }
```

This signature allows test code to elide supplying a function signature that takes the `CancellationToken` parameter and wraps it in a closure that effectively ignores it.

Reply:
I'm seeing this comment doubled as a as a comment to itself. I'm afraid to delete it in case it's just a UI error, but if it's really doubled I don't know why/how that happened and apologize for the extra noise.

Reply:
Sorry missed this one... sounds like a good idea though. I'll get this in the next round of fixes. (To be honest it slightly bothers me that TestFlow is in the builder package... the argument is that everyone has to write unit test anyhow, I guess that's reasonable.)